cg/apps/bam_to_cram.py

```diff
@@ -0,0 +1,86 @@
+"""
+    Module for compressing bam to cram
+"""
+
+import logging
+import tempfile
+from pathlib import Path
+
+from cg.utils import Process
+
+LOG = logging.getLogger(__name__)
+
+
+REFERENCE_PATH = "path/to/reference"
+SBATCH_SCRIPT_TEMPLATE = """
+#!/bin/bash
+#SBATCH -J {job_name}
+#SBATCH -A production # Maybe this should not be hardcoded if we want to test on stage
+#SBATCH -n 1 ...
+$SBATCH --mail ...
+$SBATCH --stderr ...
+$SBATCH --stdout ...
+
+source activate P_samtools # Put samtools in container and run from conda env?
+
+samtools view -c -o {cram_path} -r {reference_path} {bam_path}
+samtools index {cram_path} {cram_path}.crai
+PRE_CHECKSUM=$(samtools view {bam_path} | md5sum)
+POST_CHECKSUM=$(samtools view {cram_path} | md5sum)
+
+if [[ $PRE_CHECKSUM == $POST_CHECKSUM ]]
+then
+    echo "Compression completed"
+    touch {cram_dir}/{job_name}_cram_complete
```

Comment:
if compression is successful a file named `<sample_id>_cram_completed` is created in the same dir where the original bam-file exists. This file will serve as a flag, confirming that the compression was successful. If successful, the bam file should be deleted and removed in hk, and the cram file should be added instead. Same for bam_file field in scout.

Reply:
could it be named <sample_id>_cram_completed.txt?